packages/rum-core/src/domain/foregroundContexts.spec.ts

```diff
@@ -0,0 +1,240 @@
+import { RelativeTime, relativeNow, toServerDuration, Duration } from '@datadog/browser-core'
+import { setup, TestSetupBuilder } from '../../test/specHelper'
+import { createNewEvent } from '../../../core/test/specHelper'
+import { InForegroundPeriod } from '../rawRumEvent.types'
+import { startForegroundContexts, ForegroundContexts } from './foregroundContexts'
+
+const FOCUS_PERIOD_LENGTH = 10 as Duration
+const BLUR_PERIOD_LENGTH = 5 as Duration
+
+describe('foreground context', () => {
+  let foregroundContext: ForegroundContexts
+  let setupBuilder: TestSetupBuilder
+
+  beforeEach(() => {
+    setupBuilder = setup()
+      .withFakeClock()
+      .withConfiguration({ isEnabled: () => true })
+      .beforeBuild(({ configuration }) => {
+        foregroundContext = startForegroundContexts(configuration)
+        return foregroundContext
+      })
+  })
+
+  afterEach(() => {
+    setupBuilder.cleanup()
+  })
+
+  describe('when the page do not have the focus when starting', () => {
+    beforeEach(() => {
+      spyOn(Document.prototype, 'hasFocus').and.callFake(() => false)
+    })
+    describe('without any focus not blur event', () => {
+      describe('getInForeground', () => {
+        it('should false', () => {
+          const { clock } = setupBuilder.build()
+
+          clock.tick(1_000)
+
+          expect(foregroundContext.getInForeground(relativeNow())).toEqual(false)
+        })
+      })
+
+      describe('getInForegroundPeriods', () => {
+        it('should an empty array', () => {
+          const { clock } = setupBuilder.build()
+
+          clock.tick(1_000)
+
+          expect(foregroundContext.getInForegroundPeriods(relativeNow(), 0 as Duration)).toEqual([])
+        })
+      })
+    })
+
+    describe('with two closed focus period & one active one', () => {
+      /* 
+      events         F      B   F       B   F
+      periods        <------>   <------->   <---- - - -
+      time       0   5  10  15  20  25  30  35  40  45
+      */
+      beforeEach(() => {
+        const { clock } = setupBuilder.build()
+        clock.tick(BLUR_PERIOD_LENGTH)
+        window.dispatchEvent(createNewEvent('focus'))
+        clock.tick(FOCUS_PERIOD_LENGTH)
+        window.dispatchEvent(createNewEvent('blur'))
+        clock.tick(BLUR_PERIOD_LENGTH)
+        window.dispatchEvent(createNewEvent('focus'))
+        clock.tick(FOCUS_PERIOD_LENGTH)
+        window.dispatchEvent(createNewEvent('blur'))
+        clock.tick(BLUR_PERIOD_LENGTH)
+        window.dispatchEvent(createNewEvent('focus'))
+        clock.tick(FOCUS_PERIOD_LENGTH)
+      })
+
+      describe('getInForeground', () => {
+        it('should return false in the first blurred period', () => {
+          expect(foregroundContext.getInForeground(2 as RelativeTime)).toEqual(false)
+        })
+
+        it('should return true in the first focused period', () => {
+          expect(foregroundContext.getInForeground(10 as RelativeTime)).toEqual(true)
+        })
```

Comment:
cf https://github.com/DataDog/browser-sdk/pull/854#discussion_r635348898
what about replacing the test description by comments:
```suggestion
          // first blurred period
          expect(foregroundContext.getInForeground(2 as RelativeTime)).toEqual(false)
         // first focused period
          expect(foregroundContext.getInForeground(10 as RelativeTime)).toEqual(true)

```

Reply:
as explained before, with this approach, the test description is a duplication of the expectation.

> Label are shown in the run test providing a bit of documentation contrary to comment

IMO, it adds noise on the file and the label will only be shown when the test fails, so instead locating the expectation and the related comment in the source code should be fairly simple.  

I'd be in favor of either:

- having a single test by case you want to test, clearer than a scenario with a lot of expectations
- having a single test with a scenario and all the wanted expectations, less boilerplate and duplication